core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -1009,6 +1003,33 @@ public void handleException(TransportException exp) {
         }
     }
 
+    private ZenPing.PingCollection pingAndWait(TimeValue timeout) {
+        final ZenPing.PingCollection response = new ZenPing.PingCollection();
+        final CountDownLatch latch = new CountDownLatch(1);
+        final AtomicBoolean counted = new AtomicBoolean();
```

Comment:
nit: we can double countDown() the latch, so I don't think we need the counted boolean?